assets/app/scripts/controllers/pods.js

```diff
@@ -20,8 +20,48 @@ angular.module('openshiftConsole')
     $scope.emptyMessage = "Loading...";
     var watches = [];
 
+    $scope.gotoContainerView = function(container, jolokiaUrl) {
+      // console.log("OPENSHIFT_CONFIG: ", OPENSHIFT_CONFIG);
+      // console.log("Clicked! container: ", container, " jolokiaUrl: ", jolokiaUrl);
+      var returnTo = window.location.href;
+      var title = Core.pathGet(container, ['name']) || 'Untitled Container';
+      var targetURI = new URI();
+      targetURI.path('/console/java/')
+               .query({
+                 jolokiaUrl: jolokiaUrl,
+                 title: title,
+                 returnTo: returnTo
+               });
+      // console.log("TargetURI: ", targetURI.toString());
+      window.location.href = targetURI.toString();
+    };
+
     watches.push(DataService.watch("pods", $scope, function(pods) {
       $scope.unfilteredPods = pods.by("metadata.name");
+      // Add a jolokia URL for any container that has a jolokia port
+      _.forIn($scope.unfilteredPods, function(pod, id) {
```

Comment:
It's really handy for avoiding null references when you really want to check some attribute inside an object.  Basically avoids having a bunch of nested 'if' statements.